test/integration/scheduler/queue_test.go

```diff
@@ -1290,6 +1298,640 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			wantRequeuedPods:          sets.New("pod2"),
 			enableSchedulingQueueHint: []bool{true},
 		},
+		{
+			name:         "Pod rejected with node by the VolumeBinding plugin is requeued when the Node is created",
+			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node").Label("node", "fake-node").Obj()},
+			initialPVs: []*v1.PersistentVolume{
+				st.MakePersistentVolume().Name("pv1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadOnlyMany}).
+					Capacity(v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}).
+					HostPathVolumeSource(&v1.HostPathVolumeSource{Path: "/tmp", Type: ptr.To(v1.HostPathDirectoryOrCreate)}).
+					Obj(),
+			},
+			initialPVCs: []*v1.PersistentVolumeClaim{
+				st.MakePersistentVolumeClaim().
+					Name("pvc1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteMany}).
+					Resources(v1.VolumeResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("2Mi")}}).
+					StorageClassName(ptr.To("wait-sc")).
+					Obj(),
+			},
+			pods: []*v1.Pod{
+				st.MakePod().Name("pod1").Container("image").PVC("pvc1").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
+				node := st.MakeNode().Name("fake-node2").Label("node", "fake-node").Obj()
+				if _, err := testCtx.ClientSet.CoreV1().Nodes().Create(testCtx.Ctx, node, metav1.CreateOptions{}); err != nil {
+					return nil, fmt.Errorf("failed to create node: %w", err)
+				}
+				return map[framework.ClusterEvent]uint64{framework.NodeAdd: 1}, nil
+			},
+			wantRequeuedPods:          sets.New("pod1"),
+			enableSchedulingQueueHint: []bool{true},
```

Comment:
I think most of the test cases could work with hints both enabled and disabled, could you check this?

Reply:
The test case failed. 

- PVC/Add
- PVC/Update
- StorageClass/Update
- CSIDriver/Update

However, I believe this is not an issue, as the failure occurred because disabling QueueingHint resulted in an unschedulable Pod being scheduled. For example, PVC/Add case. . 

https://github.com/kubernetes/kubernetes/pull/127946/files#diff-1bbbb4f2aca8dab9dbd8a8d959015e1091376efdce927aa206cd04ca7ec7bb57R1466

When `pvc1` is created, if QueueingHint is enabled, `wantRequeuedPods` will only contain `pod1`. However, if QueueingHint is disabled, `wantRequeuedPods` will include both `pod1` and `pod2`.

this behavior is correct.

Reply:
Can you change `enableSchedulingQueueHint` to both `{true, false}` if the test case is successful without QHint? For example, this Node/Add case should be successful in both because it doesn't have QHint in the first place.